src/uct/rocm/copy/rocm_copy_ep.c

```diff
@@ -155,22 +167,86 @@ ucs_status_t uct_rocm_copy_ep_put_short(uct_ep_h tl_ep, const void *buffer,
                                         unsigned length, uint64_t remote_addr,
                                         uct_rkey_t rkey)
 {
-    uct_rocm_memcpy_h2d((void *)remote_addr, buffer, length);
+    uct_rocm_copy_ep_t *ep       = ucs_derived_of(tl_ep, uct_rocm_copy_ep_t);
+    uct_rocm_copy_iface_t *iface = ucs_derived_of(tl_ep->iface, uct_rocm_copy_iface_t);
+    ucs_status_t status          = UCS_OK;
+    void* base_dev_addr          = NULL;
+    uct_iov_t *iov;
+
+    if (length <= iface->config.short_h2d_thresh) {
+        uct_rocm_memcpy_h2d((void*)remote_addr, buffer, length);
+    } else {
+        iov = ucs_malloc(sizeof(uct_iov_t), "uct_iov_t");
+        if (iov == NULL) {
+            ucs_error("failed to allocate memory for uct_iov_t");
+            return UCS_ERR_NO_MEMORY;
+        }
+
+        status = uct_rocm_copy_cache_map_memhandle(ep->local_memh_cache,
+                                                   (uint64_t)buffer,
+                                                   length, &base_dev_addr);
+        if (status != UCS_OK) {
+            ucs_free(iov);
+            return status;
+        }
+
+        iov->buffer = base_dev_addr;
+        iov->length = length;
+        iov->count  = 1;
+        status      = uct_rocm_copy_ep_zcopy(tl_ep, remote_addr, iov, rkey, 1);
+        if (status != UCS_OK) {
+            ucs_error("Error in data transfer operation.");
```

Comment:
plz print status as well, also start with lowercase to be consistent with other prints thru the code

Reply:
done